stacks-core/Cargo.toml

```diff
@@ -0,0 +1,12 @@
+[package]
+name = "stacks-core"
+version = "0.1.0"
+edition = "2021"
+
+[dependencies]
+hex = "0.4.3"
```

Comment:
We've had a from scratch implementation of hex encoding in our codebase. Considering how ubiquitous hex is, and how much this crate is used, I believe there's no reason to have a custom implementation. This way we don't need to maintain it and probably won't need to think about it for a long time (if ever).

Let me know if you disagree.

Reply:
Disagree.  Please remove it.  I'd rather have the devil I know running than a devil I don't know, especially in a system that manages close to $1 billion of other peoples' money.

> Considering how ubiquitous hex is, and how much this crate is used, I believe there's no reason to have a custom implementation. This way we don't need to maintain it and probably won't need to think about it for a long time (if ever).

I strongly disagree with this mentality in the context of blockchain development when it comes to "small" dependencies like `hex`.  Code we add as a dependency is still code we're responsible for maintaining.  If it misbehaves, it hardly matters in the eyes of our users as to whether or not we wrote it, because their funds are at risk either way.  It's still our problem to fix, even if it means forking the broken package and patching it.  As such, it's better to have to debug code that we intimately understand than code that we do not.

Our hex implementation is very small compared to `hex` -- so small that it's very hard to get wrong.  It doesn't need the extra logic for `no_std` or `serde`, and we currently don't need the custom iterators that `hex` offers.